plugins/module_utils/k8s/service.py

```diff
@@ -514,7 +517,7 @@ def update(
             wait_condition = self.module.params["wait_condition"]
         results = {"changed": False, "result": {}}
 
-        if self.module.check_mode and not self.module.client.dry_run:
+        if self.module.check_mode and not self.client.dry_run:
             k8s_obj = dict_merge(existing.to_dict(), _encode_stringdata(definition))
         else:
             for merge_type in self.module.params.get("merge_type") or [
```

Comment:
The failure with the crd test is coming from here. We need to carry over the logic to test multiple merge types, since we are right now failing on the first try. So something like this in the `else` block:
```python
exception = None
for merge_type in self.module.params.get("merge_type") or ["strategic-merge", "merge",]:
    try:
        k8s_obj = self.patch_resource(...)
        exception = None
    except CoreException as e:
        exception = e
        continue
    break
if exception:
    raise exception
```

Reply:
I removed the changed I make to the crd.yml file and added this block. But it keeps failing.

```
TASK [Patch custom resource definition] ****************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed to patch object: b'{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"the body of the request was in an unknown format - accepted media types include: application/json-patch+json, application/merge-patch+json, application/apply-patch+yaml\",\"reason\":\"UnsupportedMediaType\",\"code\":415}\\n'"}
...ignoring

TASK [Assert that recreating crd is as expected] *******************************
fatal: [localhost]: FAILED! => {
    "assertion": "recreate_crd is not failed",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}
```